common-utils/src/main/java/edu/unc/lib/boxc/common/util/CLIUtil.java

```diff
@@ -0,0 +1,100 @@
+package edu.unc.lib.boxc.common.util;
+
+import edu.unc.lib.boxc.common.errors.CommandException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Utilities for executing commands
+ * @author bbpennel
+ */
+public class CLIUtil {
+    private static final Logger log = LoggerFactory.getLogger(CLIUtil.class);
+
+    private CLIUtil() {
+    }
+
+    /**
+     * Run a given command, with a timeout
+     * @param command the command to be executed
+     * @param timeout max time for the command to run, in seconds
+     * @return List containing the standard output and standard error output
+     */
+    public static List<String> executeCommand(List<String> command, long timeout) {
+        try {
+            ProcessBuilder builder = new ProcessBuilder(command);
+            Process process = builder.start();
+
+            // Use a separate thread to read the output concurrently, to avoid deadlock in case command times out
+            var outputConsumer = new StreamConsumer(process.getInputStream());
+            var errorConsumer = new StreamConsumer(process.getErrorStream());
+            outputConsumer.start();
+            errorConsumer.start();
+
+            waitForProcess(process, command, timeout);
+            // If any errors occurred while reading the output, they will be thrown here
+            outputConsumer.join();
+            errorConsumer.join();
+
+            if (process.exitValue() != 0) {
+                throw new CommandException("Command exited with errors", command,
+                        combinedOutput(outputConsumer, errorConsumer), process.exitValue());
+            }
+
+            return List.of(outputConsumer.getContent(), errorConsumer.getContent());
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throw new CommandException("Command interrupted", command, "", e);
+        } catch (IOException e) {
+            throw new CommandException("Command failed to execute", command, "", e);
+        }
+    }
+
+    private static String combinedOutput(StreamConsumer outputConsumer, StreamConsumer errorConsumer) {
```

Comment:
so this method is only used when the command exits with errors? if the above method returns a list of both, does it only add them together here since the error consumer is then not null?

Reply:
I was originally thinking I might have null checks in here, but didn't end up adding them since I think both streams will always have values, even if it's just an empty string, which should format fine. So it's mostly just making the error construction slightly shorter.

Reply:
I guess what i'm trying to ask is, since in `executeCommand` it's a list of the outputs, this method adds them together just so it's one argument in the `CommandException` method?